pkg/cmd/flagtypes/addr.go

```diff
@@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
 
 	if !a.AllowPrefix {
 		addr.Path = ""
+	} else if addr.Path == "/" {
```

Comment:
In login yes, but this is flagtypes.Addr which is shared by everything.

----- Original Message -----
> > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> >  
> >  	if !a.AllowPrefix {
> >  		addr.Path = ""
> > +	} else if addr.Path == "/" {
> 
> It fails before the config writing process (when exchanging the token,
> checking current user, etc). So I probably agree, but we still need to
> normalize it here in login too.
> 
> ----- Original Message -----
> > > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> > >  
> > >  	if !a.AllowPrefix {
> > >  		addr.Path = ""
> > > +	} else if addr.Path == "/" {
> > 
> > The task of normalizing user input doesn't belong to flagtype.Addr.  It
> > belongs to code that does checks for normalization prior to writing config.
> > 
> > ----- Original Message -----
> > > > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> > > >  
> > > >  	if !a.AllowPrefix {
> > > >  		addr.Path = ""
> > > > +	} else if addr.Path == "/" {
> > > 
> > > Because of the original issue -
> > > https://github.com/openshift/origin/issues/2358. It happens exactly when
> > > there is a trailing slash in the host.
> > > 
> > > ----- Original Message -----
> > > > > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> > > > >  
> > > > >  	if !a.AllowPrefix {
> > > > >  		addr.Path = ""
> > > > > +	} else if addr.Path == "/" {
> > > > 
> > > > Hrm - I don't really want to clear it here - what if someone else needs
> > > > it?
> > > > Why clear it here?
> > > > 
> > > > ----- Original Message -----
> > > > > > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> > > > > >  
> > > > > >  	if !a.AllowPrefix {
> > > > > >  		addr.Path = ""
> > > > > > +	} else if addr.Path == "/" {
> > > > > 
> > > > > url Parse doesn't change it, so it's only there if it's in the
> > > > > original
> > > > > string.
> > > > > 
> > > > > ----- Original Message -----
> > > > > > > @@ -106,6 +106,8 @@ func (a *Addr) Set(value string) error {
> > > > > > >  
> > > > > > >  	if !a.AllowPrefix {
> > > > > > >  		addr.Path = ""
> > > > > > > +	} else if addr.Path == "/" {
> > > > > > 
> > > > > > The behavior of url Parse is to always set this I believe.  I'm not
> > > > > > sure
> > > > > > we
> > > > > > should be clearing it here.
> > > > > > 
> > > > > > ---
> > > > > > Reply to this email directly or view it on GitHub:
> > > > > > https://github.com/openshift/origin/pull/2387/files#r30744095
> > > > > 
> > > > > 
> > > > > ---
> > > > > Reply to this email directly or view it on GitHub:
> > > > > https://github.com/openshift/origin/pull/2387/files#r30744322
> > > > 
> > > > 
> > > > ---
> > > > Reply to this email directly or view it on GitHub:
> > > > https://github.com/openshift/origin/pull/2387/files#r30744445
> > > 
> > > 
> > > ---
> > > Reply to this email directly or view it on GitHub:
> > > https://github.com/openshift/origin/pull/2387/files#r30744663
> > 
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/openshift/origin/pull/2387/files#r30744789
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2387/files#r30745697